cms/core/blocks/related.py

```diff
@@ -1,44 +1,62 @@
 from typing import TYPE_CHECKING, Any
 
 from django.utils.text import slugify
-from django.utils.translation import gettext as _
-from wagtail.blocks import CharBlock, ListBlock
+from django.utils.translation import gettext_lazy as _
+from wagtail.blocks import CharBlock, ChoiceBlock, DateBlock, ListBlock
+
+from cms.core.enums import RelatedContentType
 
 from .base import LinkBlock
 
 if TYPE_CHECKING:
+    from django_stubs_ext import StrOrPromise
     from wagtail.blocks.list_block import ListValue
 
 
 class RelatedContentBlock(LinkBlock):
     """Related content block with page or link validation."""
 
     description = CharBlock(required=False)
+    content_type = ChoiceBlock(
+        choices=RelatedContentType.choices,
+        default=RelatedContentType.ARTICLE,
+        help_text="Select the type of related content.",
+    )
+    release_date = DateBlock(required=False)
```

Comment:
note that the `ReleaseCalendarRelatedLinksStoryBlock` allows internal or external links too. and adding those currently fails with `attribute name must be string, not 'NoneType'` (`cms/core/utils.py, line 78, in get_related_content_type_label`)

Reply:
I fixed the first issue here [66c0e8d](https://github.com/ONSdigital/dis-wagtail/pull/253/commits/66c0e8da8353cdfad21d9cb0a24db284d6429382).

Regarding the second comment, would the solution be to use `RelatedLinksBlock` there for output consistency like I did in [5e4c71e](https://github.com/ONSdigital/dis-wagtail/pull/253/commits/5e4c71e0d8d22dad966da16b26bd75dc0817a58f)?

Reply:
While tempting, those links on the RC page do not include the additional metadata (type/date) - e.g. https://www.ons.gov.uk/releases/businessinvestmentintheukjanuarytomarch2025revisedresults

Reply:
Yes, much better, thank you